src/Explorer/Menus/CommandBar/ConnectionStatusComponent.tsx

```diff
@@ -78,30 +110,70 @@ export const ConnectionStatus: React.FC<Props> = ({ container }: Props): JSX.Ele
-          <span className={connectionInfo.status === ConnectionStatusType.Failed ? "connectionStatusFailed" : ""}>
-            {connectionInfo.status}
-          </span>
-          {connectionInfo.status === ConnectionStatusType.Connecting && isActive && (
-            <ProgressIndicator description={minute + ":" + second} />
-          )}
-          {connectionInfo.status === ConnectionStatusType.Connected && !isActive && (
-            <ProgressIndicator
-              className={usedGB / totalGB > 0.8 ? "lowMemory" : ""}
-              description={usedGB.toFixed(1) + " of " + totalGB.toFixed(1) + " GB"}
-              percentComplete={usedGB / totalGB}
-            />
-          )}
-        </Stack>
+    <>
+      <TooltipHost
+        content={
+          containerInfo.status &&
+          containerInfo.status === ConatinerStatusType.Active &&
+          Math.round(containerInfo.durationLeftInMinutes) <= 10
+            ? `Connected to temporary workspace. This temporary workspace will get disconnected in ${Math.round(
+                containerInfo.durationLeftInMinutes
+              )} minutes.`
+            : toolTipContent
+        }
+      >
+        <ActionButton
+          id={buttonId}
+          className={connectionInfo.status === ConnectionStatusType.Failed ? "commandReactBtn" : "connectedReactBtn"}
+          onClick={(e: React.MouseEvent<HTMLSpanElement>) =>
+            connectionInfo.status === ConnectionStatusType.Failed ? container.allocateContainer() : e.preventDefault()
+          }
+        >
+          <Stack className="connectionStatusContainer" horizontal>
+            <i className={statusColor}></i>
+            <span className={connectionInfo.status === ConnectionStatusType.Failed ? "connectionStatusFailed" : ""}>
+              {connectionInfo.status}
+            </span>
+            {connectionInfo.status === ConnectionStatusType.Connecting && isActive && (
+              <ProgressIndicator description={minute + ":" + second} />
+            )}
+            {connectionInfo.status === ConnectionStatusType.Connected && !isActive && (
+              <ProgressIndicator
+                className={usedGB / totalGB > 0.8 ? "lowMemory" : ""}
+                description={usedGB.toFixed(1) + " of " + totalGB.toFixed(1) + " GB"}
+                percentComplete={usedGB / totalGB}
+              />
+            )}
+          </Stack>
+          {!isBarDismissed &&
+          containerInfo.status &&
+          containerInfo.status === ConatinerStatusType.Active &&
+          Math.round(containerInfo.durationLeftInMinutes) <= 10 ? (
+            <FocusTrapCallout
+              role="alertdialog"
+              className={styles.callout}
+              gapSpace={0}
+              target={`#${buttonId}`}
+              onDismiss={() => setIsBarDismissed(true)}
+              setInitialFocus
+            >
+              <Text block variant="xLarge" className={styles.title}>
+                Remaining Time
```

Comment:
@swviswan /@deborahc  Can you please suggest title for this Callout

Reply:
Remaining time